### PR TITLE
Migrate cmd/kubelet and pkg/kubelet/cadvisor , pkg/kubelet/cri/remote/util , pkg/kubelet/images to structured logging

### LINKED ISSUE #98976: Kubelet migration to structured logs
For Structured Logging Migration: https://github.com/kubernetes/enhancements/issues/1602

[Beta graduation criteria:](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#beta)
- Structured logging interface design is verify by migrating one whole component (Kubelet) to structured logging. All other component migrations will be best effort.

### What would you like to be added:

Please migrate large chunks of code in each PR, at **minimum** one top level directory at a time (assuming your diff does not exceed ~40 lines). There are already ~40 top-level directories in the kubelet. If we have fewer PRs, it will be much easier for reviewers and approvers to merge your change.

The following query can be used to identify directories that still need structured logging migration:

```bash
git grep -L klog pkg/kubelet/ | sed 's|[^/]\+$||' | sort -u
git grep -L klog cmd/kubelet/ | sed 's|[^/]\+$||' | sort -u
```

You can check the project board to see if someone has already migrated a certain file: https://github.com/orgs/kubernetes/projects/53

#### Authors: What to include in your PR

- **Subject:** "Migrate `<directory>` to structured logging"
- **SIGs:** instrumentation, node
- **Priority:** important-soon
- **Kind:** cleanup
- **Release note:** You **must** include a release note for any log changes!

**HOWTO:** Read the [structured logging migration guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md).

_Only_ reference this issue for kubelet migrations. Don't mark it as "fixes" or other keywords it will close this tracker. All other components are best effort for 1.21 and should have priority/backlog.

Before you submit your PR, check to see if someone has already migrated that component: https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+structured+log+label%3Asig%2Fnode

#### Reviewers: What to look for

Read through the migration guide above. Use the PR query above to find PRs to review. Follow standard [node triage guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/triage.md).

### Why is this needed:

See above.

/sig instrumentation
/sig node
/cc @serathius 
/milestone v1.21

### PR DESCRIPTION
**Pull request by LittleGoWitch**
#### What type of PR is this?

/kind cleanup
/sig instrumentation
/sig node

#### What this PR does / why we need it:
Migrate `cmd/kubelet/`+ `pkg/kubelet/cadvisor/cadvisor_linux.go` + `pkg/kubelet/cri/remote/util/util_unix.go` + `pkg/kubelet/images/image_manager.go` to structured logging
#### Special notes for your reviewer:
part of [kubernetes/enhancements#1602](https://github.com/kubernetes/enhancements/issues/1602)
As it is related to log, change the log here to structured log.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- [KEP]: [Structured Logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)

```docs

```

### CODE REVIEW

File: pkg/kubelet/images/image_manager.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 	logPrefix := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, container.Image)
 	ref, err := kubecontainer.GenerateContainerRef(pod, container)
 	if err != nil {
-		klog.Errorf("Couldn't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
+		klog.ErrorS(err, "Couldn't make a ref to pod", "pod", klog.KObj(pod), "container", container.Name)
```
Comment (serathius) [inline pkg/kubelet/images/image_manager.go:93]:
```suggestion
		klog.ErrorS(err, "Couldn't make a ref to pod", "pod", klog.KObj(pod), "containerName", container.Name)
```

**Review by ehashman — COMMENTED** [review]
/lgtm
/priority important-soon
/triage accepted

**Review by mrunalp — APPROVED** [review]

**Pull request merged**